### ISSUE TITLE
Upcoming major version bump

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#308]
Since we use CalVer, we have one chance a year to make breaking changes: the first release after the new year. Does anybody have any breaking/backwards-incompatible changes they'd like to make for the major version bump?

Things I'd like to do:
- #269.
- I'd also add `typing.Match` and `typing.Pattern` to the imports forbidden by Y022 (Python 3.6 is now long EOL, and unsupported by at least typeshed and mypy).

**Comment by JelleZijlstra** [comment]
I don't think we should think of the new year as a major version bump. We don't do semantic versioning and breaking changes may appear in any version, though we should handle them in a way that's friendly to users (e.g., with deprecation warnings).

That said, I'm supportive of the changes you're suggesting.

**Comment by AlexWaygood** [comment]
I'm definitely not in a rush to make breaking changes with no good reason :)
